osu.Game/OsuGame.cs

```diff
@@ -792,7 +792,7 @@ public override Task Import(ImportTask[] imports, ImportParameters parameters =
 
         protected virtual UpdateManager CreateUpdateManager() => new UpdateManager();
 
-        protected virtual HighPerformanceSession CreateHighPerformanceSession() => new HighPerformanceSession();
+        protected virtual HighPerformanceSessionManager CreateHighPerformanceSessionManager() => new HighPerformanceSessionManager();
```

Comment:
Was there a reason for this being `virtual`? Mobile or something? If not required, then:

```diff
diff --git a/osu.Game/OsuGame.cs b/osu.Game/OsuGame.cs
index f23a78d2dc..13e80e0707 100644
--- a/osu.Game/OsuGame.cs
+++ b/osu.Game/OsuGame.cs
@@ -792,8 +792,6 @@ public override Task Import(ImportTask[] imports, ImportParameters parameters =
 
         protected virtual UpdateManager CreateUpdateManager() => new UpdateManager();
 
-        protected virtual HighPerformanceSessionManager CreateHighPerformanceSessionManager() => new HighPerformanceSessionManager();
-
         protected override Container CreateScalingContainer() => new ScalingContainer(ScalingMode.Everything);
 
         #region Beatmap progression
@@ -1088,7 +1086,7 @@ protected override void LoadComplete()
             loadComponentSingleFile(new AccountCreationOverlay(), topMostOverlayContent.Add, true);
             loadComponentSingleFile<IDialogOverlay>(new DialogOverlay(), topMostOverlayContent.Add, true);
 
-            loadComponentSingleFile(CreateHighPerformanceSessionManager(), Add, true);
+            loadComponentSingleFile(new HighPerformanceSessionManager(), Add, true);
 
             loadComponentSingleFile(new BackgroundDataStoreProcessor(), Add);
 
diff --git a/osu.Game/Performance/HighPerformanceSessionManager.cs b/osu.Game/Performance/HighPerformanceSessionManager.cs
index c5146933b7..22c929083c 100644
--- a/osu.Game/Performance/HighPerformanceSessionManager.cs
+++ b/osu.Game/Performance/HighPerformanceSessionManager.cs
@@ -14,11 +14,11 @@ public partial class HighPerformanceSessionManager : Component
 
         public IDisposable BeginSession()
         {
-            EnableHighPerformanceSession();
-            return new InvokeOnDisposal<HighPerformanceSessionManager>(this, static m => m.DisableHighPerformanceSession());
+            enableHighPerformanceSession();
+            return new InvokeOnDisposal<HighPerformanceSessionManager>(this, static m => m.disableHighPerformanceSession());
         }
 
-        protected virtual void EnableHighPerformanceSession()
+        private void enableHighPerformanceSession()
         {
             originalGCMode = GCSettings.LatencyMode;
             GCSettings.LatencyMode = GCLatencyMode.LowLatency;
@@ -27,7 +27,7 @@ protected virtual void EnableHighPerformanceSession()
             GC.Collect(0);
         }
 
-        protected virtual void DisableHighPerformanceSession()
+        private void disableHighPerformanceSession()
         {
             if (GCSettings.LatencyMode == GCLatencyMode.LowLatency)
                 GCSettings.LatencyMode = originalGCMode;

```

Reply:
as long as it's not an `OperationNotSupported` 😅